### PR TITLE
[wasm] Initial netcore support.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
This is initial support for wasm+netcore, controlled by the ENABLE_WASM_NETCORE make flag in Make.config. This is an optional build configuration which can work alongside a normal build.

**Comment by vargaz** [comment]
See the run-hello-netcore target for usage.

**Comment by vargaz** [comment]
Note that this cannot be enabled on CI, since it depends on an external corefx repo to compile a wasm version of System.Native plus some System.Native patches.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

File: sdks/wasm/netcore-corefx.diff [hunk 1]
```diff
@@ -0,0 +1,55 @@
+diff --git a/src/Native/Unix/CMakeLists.txt b/src/Native/Unix/CMakeLists.txt
```
Comment (marek-safar) [inline sdks/wasm/netcore-corefx.diff]: Why is this needed? We should not be building this under Mono

Comment (vargaz) [reply sdks/wasm/netcore-corefx.diff]: There are no wasm binaries for System.Native.

Comment (filipnavara) [reply sdks/wasm/netcore-corefx.diff]: We should bring them back and publish through NuGet. That is no trivial effort though. At some point the WASM targets existed in CoreFX but they were removed and never made it to Azure Pipelines. I've been told that there should be a machine pool with emscripten and other relevant tools installed but I know no details about that.

Comment (marek-safar) [reply sdks/wasm/netcore-corefx.diff]: I agree we need to start the effort on corefx side as well (I'll look into that)

Comment (filipnavara) [reply sdks/wasm/netcore-corefx.diff]: For reference, the initial WebAssembly support was added with https://github.com/dotnet/corefx/pull/28936. Various parts were removed by https://github.com/dotnet/corefx/pull/36739.

Comment (filipnavara) [reply sdks/wasm/netcore-corefx.diff]: Here's the definition for the Azure Pipelines pool:
```
        pool:
            name:  NetCorePublic-Pool
            queue: buildpool.windows.10.wasm.open
```

Comment (jkotas) [reply sdks/wasm/netcore-corefx.diff]: > Various parts were removed by dotnet/corefx#36739

The initial WebAssembly support was on top of ProjectN. The removed parts were the ProjectN specific plumbing.

**Comment by vargaz** [comment]
System.Native needs to be built with the same emscripten version we are using, the same flags, etc. We might need more than one version depending on flags.
